src/bin/solang.rs

```diff
@@ -301,10 +307,13 @@ fn main() {
         }
     } else {
         let opt_level = match matches.value_of("OPT").unwrap() {
-            "none" => OptimizationLevel::None,
-            "less" => OptimizationLevel::Less,
-            "default" => OptimizationLevel::Default,
-            "aggressive" => OptimizationLevel::Aggressive,
+            "0" => OptimizationLevel::None,
+            "1" => OptimizationLevel::Less,
+            "2" => OptimizationLevel::Default,
+            // According to https://llvm.org/doxygen/CodeGen_8h_source.html -O2 and -Os are default
+            "s" => OptimizationLevel::Default,
+            "S" => OptimizationLevel::Default,
```

Comment:
Here is where the change falls apart. We need to tell llvm about the size level, so we need to have a `opt_speed_level` and `opt_size_level`. This needs to be set in the `Options` struct, and then used in `src/emit/binary.rs` to set the llvm size/speed optimization level. See here the inkwell call for this https://github.com/TheDan64/inkwell/blob/master/src/passes.rs#L100-L102